shared/src/commonMain/kotlin/co/nimblehq/avishek/phong/kmmic/presentation/module/HomeViewModel.kt

```diff
@@ -18,56 +14,78 @@ private const val DEFAULT_PAGE_SIZE = 5
 data class HomeViewState(
     val isLoading: Boolean,
     val headerUiModel: SurveyHeaderUiModel? = null,
-    var surveys: List<SurveyUiModel> = listOf()
+    var surveys: List<SurveyUiModel> = listOf(),
 ) {
-    constructor() : this (true)
+    constructor() : this(true)
 }
 
 class HomeViewModel(
     private val getUserProfileUseCase: GetUserProfileUseCase,
     private val getSurveysUseCase: GetSurveysUseCase,
+    private val getAppVersionUseCase: GetAppVersionUseCase,
     private val dateTime: DateTime,
     private val dateTimeFormatter: DateTimeFormatter,
 ) : BaseViewModel() {
 
     private val _viewState: MutableStateFlow<HomeViewState> = MutableStateFlow(HomeViewState())
-
     val viewState: StateFlow<HomeViewState> = _viewState
 
+    private val _appVersion: MutableStateFlow<String> = MutableStateFlow("")
+    val appVersion: StateFlow<String> = _appVersion
+
     private var previousSelectedIndex = 0
     private var currentPage = 1
 
-    fun fetchData() {
+    init {
+        fetchData()
+    }
+
+    private fun fetchData() {
```

Comment:
@AVI5HEK, would you make this function as public? I'm asking because it is failing to build the iOS module. I'll update the code base on the iOS module after this get merged.

Reply:
Alrighty.